mitosheet/dev/create_new_step.py

```diff
@@ -0,0 +1,379 @@
+            type = input(f'Select a type from {VALID_PARAMETER_TYPES} for {param_name}: ')
+        params[param_name] = type
+    
+    return params
+
+def get_test_file_code(original_step_name: str, params: Dict[str, str]) -> str:
+    step_name = get_step_name(original_step_name)
+    params_string = ", ".join(params.keys())
+
+    return f"""#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+\"\"\"
+Contains tests for {original_step_name}
+\"\"\"
+
+import pandas as pd
+import pytest
+from mitosheet.tests.test_utils import create_mito_wrapper_dfs
+
+{step_name.upper()}_TESTS = [
+    (
+        [
+            pd.DataFrame({OPEN_BRACKET}'A': [1, 2, 3]{CLOSE_BRACKET})
+        ],
+        "put", 
+        "params", 
+        "here",
+        [
+            pd.DataFrame({OPEN_BRACKET}'A': [1, 2, 3]{CLOSE_BRACKET})
+        ]
+    ),
+]
+@pytest.mark.parametrize(\"input_dfs, {params_string}, output_dfs\", {step_name.upper()}_TESTS)
+def test_fill_na(input_dfs, {params_string}, output_dfs):
+    mito = create_mito_wrapper_dfs(*input_dfs)
+
+    mito.{step_name}({params_string})
+
+    assert len(mito.dfs) == len(output_dfs)
+    for actual, expected in zip(mito.dfs, output_dfs):
+        assert actual.equals(expected)"""
+
+def write_testing_file(original_step_name: str, params: Dict[str, str]) -> None:
+    step_name = get_step_name(original_step_name)
+    path_to_file = get_test_folder() / ("test_" + step_name + '.py') 
+    test_file_code = get_test_file_code(original_step_name, params)
+    write_python_code_file(path_to_file, test_file_code)
+    
+
+def main() -> None:
+    """
+    This bit of meta-programming makes it easy to quickly generate the barebones
+    of a new step type, adding files in the right locations, etc. In the future,
+    we can build onto to this to make it much quicker to add new steps, which
+    will be a nice thing for us.
+    """
+    print("Welcome to the Mito step generator. This will walk you through the process of generating a new step...")
+    original_step_name = input("Step Name: [Fill NaN, Drop Duplicates] ")
+    params = read_params()
+
+    # First, we change all the Python files
+    write_step_performer(original_step_name, params)
+    print("Wrote step performer")
+    write_code_chunk(original_step_name, params)
+    print("Wrote code chunk")
+    write_to_step_performer_init(original_step_name)
+    print("Wrote to init")
+    write_to_test_utility(original_step_name, params)
+    print("Wrote to test utilities")
+    write_testing_file(original_step_name, params)
+    print("Wrote testing file")
+
+    # TODO:
+    # 1. Add a new step type in Types.tsx
+    # 2. Auto generate the api files
+    # 3. Ask about each of the params
+    # 4. Be able to read the params back in correctly (and do some refactoring!)
+    # 5. Be able to do upgrade refactoring automatically - just tell it how to migrate
+
+
+
```

Comment:
Want to add a short read me to the dev folder that just at least explains what the folder is and what is in it, and how to run the commands.